isos/appliance-staging.sh

```diff
@@ -67,73 +66,48 @@ unpack $PACKAGE $PKGDIR
 # Below: the image authoring
 #################################################################
 
-# Install VCH base packages
-#
-# List stable packages here
-#   e2fsprogs # for mkfs.ext4
-#   procps-ng # for ps
-#   iputils   # for ping
-#   iproute2  # for ip
-#   iptables  # for iptables
-#   net-tools # for netstat
-#   openssh   # for ssh server
-#   sudo      # for sudo
-#
-# Temporary packages list here
-#   systemd   # for convenience only at this time
-#   tndf      # so we can deploy other packages into the appliance live - MUST BE REMOVED FOR SHIPPING
-#   vim       # basic editing function
-#   lsof      # for debugging issues unmounting disks for the copy/diff paths
-yum_cached -c $cache -u -p $PKGDIR install \
-    haveged \
-    systemd \
-    openssh \
-    iptables \
-    e2fsprogs \
-    procps-ng \
-    iputils \
-    iproute2 \
-    iptables \
-    net-tools \
-    sudo \
-    tdnf \
-    vim \
-    gzip \
-    lsof \
-    logrotate \
-    photon-release \
-    -y --nogpgcheck
-
-# https://www.freedesktop.org/wiki/Software/systemd/InitrdInterface/
-touch $(rootfs_dir $PKGDIR)/etc/initrd-release
+# Install VCH packages
+# load the repo to use from the package if not explicit in env
+REPO=${REPO:-$(cat $PKGDIR/repo.cfg)}
+REPODIR="$DIR/base/repos/${REPO}/"
+PACKAGE_MANAGER=${PACKAGE_MANAGER:-$(cat $REPODIR/repo-spec.json | jq -r '.packagemanager')}
+PACKAGE_MANAGER=${PACKAGE_MANAGER:-tdnf}
+setup_pm $REPODIR $PKGDIR $PACKAGE_MANAGER $REPO
+
+script=$(cat $REPODIR/repo-spec.json | jq -r '.packages_script_staging.appliance')
+[ -n "$script" ] && package_cached -c $cache -u -p $PKGDIR $script --nogpgcheck -y
+
+STAGING_PKGS=$(cat $REPODIR/repo-spec.json | jq -r '.packages.appliance')
+package_cached -c $cache -u -p $PKGDIR install $STAGING_PKGS --nogpgcheck -y
 
 # Give a permission to vicadmin to run iptables.
 echo "vicadmin ALL=NOPASSWD: /sbin/iptables --list" >> $(rootfs_dir $PKGDIR)/etc/sudoers
 
 # ensure we're not including a cache in the staging bundle
 # but don't update the cache bundle we're using to install
-yum_cached -p $PKGDIR clean all
+package_cached  -p $PKGDIR clean all
 
 # configure us for autologin of root
-#COPY override.conf $ROOTFS/etc/systemd/system/getty@.service.d/
-# HACK until the issues with override.conf above are dealt with
-pwhash=$(openssl passwd -1 -salt vic password)
-sed -i -e "s/^root:[^:]*:/root:${pwhash}:/" $(rootfs_dir $PKGDIR)/etc/shadow
+getty_dir=$(rootfs_dir $PKGDIR)/usr/lib/systemd/system/getty@tty1.service.d/
+mkdir -p $getty_dir
+cp ${DIR}/appliance/override.conf $getty_dir
```

Comment:
I'm curious how this functions with the shell being set to /bin/false below. We need to keep the shell disabled facet but would be nice to autologin once `vic-machine debug` updates the password file to allow it.

Reply:
I've confirmed that, when set to /bin/false, the getty logs on correctly but does not display a shell. Once vic-machine debug enables the shell, the shell will properly display it after a `ctrl^c` to quit the current getty session.